packages/core/src/application.ts

```diff
@@ -61,9 +63,83 @@ export class Application extends Context {
    */
   controller(controllerCtor: ControllerClass, name?: string): Binding {
     name = name || controllerCtor.name;
-    return this.bind(`controllers.${name}`)
+    return this.bind(`${CoreBindings.CONTROLLERS_PREFIX}.${name}`)
       .toClass(controllerCtor)
-      .tag('controller');
+      .tag(CoreBindings.CONTROLLERS_TAG);
+  }
+
+  /**
+   * Register a booter class / array of classes with this application.
+   *
+   * @param {Function | Function[]} booterCls The booter class (constructor function).
+   * @param {string=} name Optional booter name, defaults to the class name.
+   * Ignored is cls is an Array and the name defaults to the class name.
+   * @returns {Binding | Binding[]} The newly created binding(s), you can use the
+   * reference to further modify the binding, e.g. lock the value to prevent
+   * further modifications.
+   *
+   * ```ts
+   * class MyBooter implements Booter {}
+   * app.booter(MyBooter);
+   * ```
+   */
+  booter(booterCls: Constructor<Booter>, name?: string): Binding;
+  booter(booterCls: Constructor<Booter>[]): Binding[];
+  booter(
+    booterCls: Constructor<Booter> | Constructor<Booter>[],
+    name?: string,
+    // tslint:disable-next-line:no-any
+  ): any {
+    if (Array.isArray(booterCls)) {
+      return booterCls.map(cls => this._bindBooter(cls));
+    } else {
+      return this._bindBooter(booterCls, name);
+    }
+  }
+
+  /**
+   *
+   * @param booterCls A Booter Class
+   * @param {string} name Name the Booter Class should be bound to
+   * @returns {Binding} The newly created Binding
+   */
+  private _bindBooter<T extends Booter>(
+    booterCls: Constructor<T>,
+    name?: string,
+  ): Binding {
+    name = name || booterCls.name;
+    return this.bind(`${CoreBindings.BOOTER_PREFIX}.${name}`)
+      .toClass(booterCls)
+      .inScope(BindingScope.CONTEXT)
+      .tag(CoreBindings.BOOTER_TAG);
+  }
+
+  /**
+   * Function is responsible for calling all registered Booter classes that
+   * are bound to the Application instance. Each phase of an instance must
+   * complete before the next phase is started.
+   *
+   * @param {BootExecutionOptions} execOptions Options to control the boot
+   * process for the Application
+   */
+  async boot(execOptions?: BootExecutionOptions): Promise<void> {
+    // Get a instance of the BootStrapper
+    const bootstrapper = await this.get(CoreBindings.BOOTSTRAPPER, {
+      optional: true,
+    });
+    // Since bootstrapper is optional, we check to see if instance was returned
+    if (bootstrapper) {
```

Comment:
We usually use reverse logic + early return to keep the code easier to read and less indented.

```ts
if (!bootsrapper) {
  // handle this error case and return or throw
}

if (!this.options) {
 // handle this error case and return or throw
}

await boostrapper.boot(/*...*/);
```